src/device-state/current-state.ts

```diff
@@ -2,55 +2,47 @@ import * as _ from 'lodash';
 import * as url from 'url';
 import { CoreOptions } from 'request';
 
-import * as constants from '../lib/constants';
+import { backoffPromise } from '../lib/backoff';
 import { log } from '../lib/supervisor-console';
 import { InternalInconsistencyError, StatusError } from '../lib/errors';
 import { getRequestInstance } from '../lib/request';
 import * as sysInfo from '../lib/system-info';
+import sleep from '../lib/sleep';
```

Comment:
Creating this instead of using bluebird's delay. I will make a commit to replace all delays with our sleep.